packages/runtime/src/helpers/config.ts

```diff
@@ -34,18 +34,52 @@ const defaultFailBuild = (message: string, { error }): never => {
   throw new Error(`${message}\n${error && error.stack}`)
 }
 
+const addIncrementalCacheHandlerPath = (
+  incrementalCacheHandlerPath: string,
+  experimentalConfig: NextConfig['experimental'],
+) => {
+  // This check is needed for now because if blob storage isn't available, this file will not have been created
+  if (incrementalCacheHandlerPath !== undefined && existsSync(incrementalCacheHandlerPath)) {
+    experimentalConfig.incrementalCacheHandlerPath = incrementalCacheHandlerPath
+  }
+
+  return experimentalConfig
+}
+
 export const getNextConfig = async function getNextConfig({
   publish,
   failBuild = defaultFailBuild,
+  incrementalCacheHandlerPath,
+}: {
+  publish: string
+  failBuild?: (message: string, { error }) => never
+  incrementalCacheHandlerPath?: string
 }): Promise<NextConfig> {
   try {
-    const { config, appDir, ignore }: RequiredServerFiles = await readJSON(join(publish, 'required-server-files.json'))
+    const requiredServerFiles: RequiredServerFiles = await readJSON(join(publish, 'required-server-files.json'))
+    const { config, appDir, ignore } = requiredServerFiles
+
     if (!config) {
       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
       // @ts-ignore
       return failBuild('Error loading your Next config')
     }
 
+    if ('incrementalCacheHandlerPath' in config.experimental) {
```

Comment:
It's unlikely someone deploying to Netlify would add a custom incremental cache handler, but I added this warning in case.